app/src/main/java/org/mozilla/focus/broadcastreceiver/DownloadBroadcastReceiver.java

```diff
@@ -0,0 +1,120 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.broadcastreceiver;
+
+import android.app.DownloadManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.design.widget.Snackbar;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.FileProvider;
+import android.view.View;
+import android.webkit.MimeTypeMap;
+import android.webkit.URLUtil;
+
+import org.mozilla.focus.BuildConfig;
+import org.mozilla.focus.R;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * BroadcastReceiver for finished downloads
+ */
+public class DownloadBroadcastReceiver extends BroadcastReceiver {
+    private static final String FILE_SCHEME = "file://";
+    private static final String FILE_PROVIDER_EXTENSION = ".fileprovider";
+    private final View browserContainer;
```

Comment:
nit: Conventionally, constants (static final) should be separated from instance variables; i.e. add a newline above the view.